internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -445,24 +453,35 @@ func (c *enrollCmd) prepareFleetTLS() error {
 
 func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	err := c.daemonReload(ctx)
+	if err != nil &&
+		(errors.Is(err, context.DeadlineExceeded) ||
+			errors.Is(err, context.Canceled)) {
+		return fmt.Errorf("could not reload daemon: %w", err)
+	}
 	if err == nil {
 		return nil
 	}
 
 	signal := make(chan struct{})
+	defer close(signal)
 	backExp := backoff.NewExpBackoff(signal, 10*time.Second, 1*time.Minute)
 
-	for i := 5; i >= 0; i-- {
+	for i := 0; i < 5; i++ {
 		backExp.Wait()
 		c.log.Info("Retrying to restart...")
 		err = c.daemonReload(ctx)
+		if err != nil &&
+			(errors.Is(err, context.DeadlineExceeded) ||
+				errors.Is(err, context.Canceled)) {
+			return fmt.Errorf("could not reload daemon after %d retries: %w",
+				i+1, err)
+		}
 		if err == nil {
-			break
+			return nil
 		}
```

Comment:
Same here (cannot suggest because of the deleted line -.- )

```
		if err == nil {
			return nil
		}
		if errors.Is(err, context.DeadlineExceeded) ||
				errors.Is(err, context.Canceled) {
			return fmt.Errorf("could not reload daemon after %d retries: %w",
				i+1, err)
		}

```